src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/ReadyToRunReader.cs

```diff
@@ -392,17 +394,48 @@ public ReadyToRunReader(IAssemblyResolver assemblyResolver, IAssemblyMetadata me
         }
 
         /// <summary>
-        /// Initializes the fields of the R2RHeader and R2RMethods
+        /// Minimally initializes the R2R reader.
         /// </summary>
-        /// <param name="filename">PE image</param>
-        /// <exception cref="BadImageFormatException">The Cor header flag must be ILLibrary</exception>
+        /// <param name="assemblyResolver">Assembly resolver</param>
+        /// <param name="metadata">Assembly metadata</param>
+        /// <param name="peReader">PE image</param>
+        /// <param name="filename">PE file name</param>
+        /// <param name="content">PE image content</param>
+        public ReadyToRunReader(IAssemblyResolver assemblyResolver, IAssemblyMetadata metadata, PEReader peReader, string filename, ImmutableArray<byte> content)
```

Comment:
Would it be more flexible to take the content as `ReadOnlyMemory<byte>`?

Reply:
> Is it possible to convert a ReadOnlyMemory<byte> to a byte[] without allocation?

It is - if the underlying storage is an array.

Reply:
Check `MemoryMarshal.TryGetArray<T>`

Also, how much the code does actually need an array? It seems like the array gets pinned and most of the code operates on the pinned memory.

Reply:
Just did a quick experiment to change the `Image` property from a `byte[]` to `ReadOnlyMemory<byte>`. 

We get around 73 errors. 

The majority of the error is due to some underlying APIs expects a `byte[]` instead of a `ReadOnlyMemory<byte>` as expected - examples include `BitConverter.ToUInt32`, `NativeReader.ReadInt32` or `Encoding.UTF8.GetString`.